packages/dds/tree/src/tree/visitDelta.ts

```diff
@@ -184,10 +184,14 @@ function secondPass(delta: Delta.MarkList, props: PassProps): void {
             const type = mark.type;
             switch (type) {
                 case Delta.MarkType.ModifyAndDelete:
-                case Delta.MarkType.Delete:
                 case Delta.MarkType.ModifyAndMoveOut:
+                    // Handled in the first pass
+                    index -= 1;
+                    break;
+                case Delta.MarkType.Delete:
                 case Delta.MarkType.MoveOut:
                     // Handled in the first pass
+                    index -= mark.count;
```

Comment:
I'm now realizing that I think the code was correct before: we only need to not increment the index as opposed to decrementing the index.